azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java

```diff
@@ -101,7 +102,7 @@
              * @param vmSize the size of each virtual machine in the agent pool
              * @return the next stage of the definition
              */
-            WithAttach<ParentT> withVirtualMachineSize(ContainerServiceVMSizeTypes vmSize);
+            WithAgentPoolVirtualMachineCount<ParentT> withVirtualMachineSize(ContainerServiceVMSizeTypes vmSize);
```

Comment:
It may be a large breaking change.

Reply:
Probable, copied from track2.

Reply:
For this one, we can revert back to `WithAttach`.

For `withAgentPoolVirtualMachineCount`, we will need to add it back to provide backward compabitility?

Reply:
I think it is better for backward compatibility. Due to maintenance, it could also do breaking change if need.

Reply:
https://github.com/Azure/azure-libraries-for-java/pull/1389/commits/c8e7737be9c5e56a8eecd4d078c1c64cfa96f1ff put them back for backward compatibility.